framework/OutStreams/PlotInterfaces/PopulationPlot.py

```diff
@@ -0,0 +1,178 @@
+      self.sourceName = params['source']
+      self.vars       = params['vars']
+      self.index      = params['index']
+      self.how        = params['how']
+
+    params, notFound = spec.findNodesAndExtractValues(['logVars'])
+    if notFound:
+      self.logVars = None
+    else:
+      self.logVars = params['logVars']
+
+
+  def initialize(self, stepEntities):
+    """
+      Function to initialize the OutStream. It basically looks for the "data"
+      object and links it to the system.
+      @ In, stepEntities, dict, contains all the Objects are going to be used in the
+                                current step. The sources are searched into this.
+      @ Out, None
+    """
+    src = self.findSource(self.sourceName, stepEntities)
+    if src is None:
+      self.raiseAnError(IOError, f'No source named "{self.sourceName}" was found in the Step for SamplePlot "{self.name}"!')
+    self.source = src
+
+    dataVars = self.source.getVars()
+    missing = [var for var in (self.vars) if var not in dataVars]
+    if missing:
+      msg = f'Source DataObject "{self.source.name}" is missing the following variables ' +\
+            f'expected by OptPath plotter "{self.name}": '
+      msg += ', '.join(f'"{m}"' for m in missing)
+      self.raiseAnError(IOError, msg)
+
+  def run(self):
+    """
+      Main run method.
+      @ In, None
+      @ Out, None
+    """
+    data = self.source.asDataset().to_dataframe()
+    inVars  = self.source._inputs
+    outVars = self.source._outputs
+
+    nFigures = len(self.vars)
+    fig, axs = plt.subplots(nFigures,1)
+    fig.suptitle('Population Plot')
+
+    min_Gen = int(min(data[self.index]))
+    max_Gen = int(max(data[self.index]))
+
+    for indexVar,var in enumerate(self.vars):
+      min_fit = np.zeros(max_Gen-min_Gen+1)
+      max_fit = np.zeros(max_Gen-min_Gen+1)
+      avg_fit = np.zeros(max_Gen-min_Gen+1)
+
+      for idx,genID in enumerate(range(min_Gen,max_Gen+1,1)):
+        population = data[data[self.index]==genID]
+        min_fit[idx] = min(population[var])
+        max_fit[idx] = max(population[var])
+        avg_fit[idx] = population[var].mean()
+
+      if var in inVars:
+        if var in self.logVars:
+          errorFill(range(min_Gen,max_Gen+1,1), avg_fit, [min_fit,max_fit], color='g', ax=axs[indexVar],logscale=True)
+        else:
+          errorFill(range(min_Gen,max_Gen+1,1), avg_fit, [min_fit,max_fit], color='g', ax=axs[indexVar])
+      else:
+        if var in self.logVars:
+          errorFill(range(min_Gen,max_Gen+1,1), avg_fit, [min_fit,max_fit], color='b', ax=axs[indexVar],logscale=True)
+        else:
+          errorFill(range(min_Gen,max_Gen+1,1), avg_fit, [min_fit,max_fit], color='b', ax=axs[indexVar])
+      axs[indexVar].set_ylabel(var)
+      if var == self.vars[-1]:
+        axs[indexVar].set_xlabel('Batch #')
+
+    if self.how in ['png','pdf','svg','jpeg']:
+      fileName = self.name +'.%s'  % self.how
+      plt.savefig(fileName, format=self.how)
```

Comment:
in the future this should be abstracted imo, probably all the way back to the Plot baseclass, so we can call something like `self.saveFigure(fig)` and the base class handles the "how" business.

Reply:
I agree, see my previous comment.